### PR TITLE
Align ST sensor to stmemsc HAL v1.03

### PR DESCRIPTION
**Pull request by avisconti**
Requires: https://github.com/zephyrproject-rtos/hal_st/pull/5

Upgrading stmemsc HAL module to v1.03 requires to align all sensor drivers using it.
Drivers have been tested on:
 - x_nucleo_iks02a1 shield
 - sensortile_box board

This PR includes also two drivers fixes found during tests:
 - drivers/sensor: iis2mdc: Fix the temperature in celsius
 - drivers/sensor: ism330dhcx: Fix init routines order

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -71,7 +71,7 @@ manifest:
       revision: 69c3e1e6e167767cb75aa2b468df4ade05988cdb
       path: modules/hal/silabs
     - name: hal_st
-      revision: 5b3ec3e182d4310e8943cc34c6c70ae57d9711da
+      revision: pull/5/head
```
Comment (MaureenHelm) [inline west.yml:74]: Adding DNM label to this PR until the hal PR is merged

**Review by MaureenHelm — COMMENTED** [review]
LGTM

**Review by MaureenHelm — APPROVED** [review]

**Comment by avisconti** [comment]
@MaureenHelm  @erwango 
Update: I have implemented/tested a new device (iis2iclx) introduced in stmemsc HAL v1.03. The new driver work is not completed yet, but I would say that the HAL v1.03 is integrated well.

@erwango 
If you could also take a look just to see if I have forgotten anything. Thanks!

**Review by erwango — APPROVED** [review]

**Comment by avisconti** [comment]
@MaureenHelm  @erwango 
Thanks for reviewing. I'm a bit in hurry because there is an external request for the iis2iclx device driver (#29129) which depends by this PR

**Pull request merged**